base/intfuncs.jl

```diff
@@ -19,8 +19,8 @@ function gcd{T<:Union{Int64,UInt64,Int128,UInt128}}(a::T, b::T)
     za = trailing_zeros(a)
     zb = trailing_zeros(b)
     k = min(za, zb)
-    u = abs(a >> za)
-    v = abs(b >> zb)
+    u = unsigned(abs(a >> za))
+    v = unsigned(abs(b >> zb))
```

Comment:
`checked_abs` is too strict here; if `a == b == typemin(Int)`, then `unsigned(abs(...))` is well-defined, but `checked_abs` would fail.

I just see that this code right-shifts before calling `abs`, and given that `typemin(Int)` has many trailing zeros, the code is probably fine.

No, they give different errors. `checked_abs` returns `OverflowError`, whereas type conversions return `InexactError`. (I find that weird, but that's how things are.) The generic gcd algorithm also might return `DivideError` instead.